src/WinGetServer/WinGetServerManualActivation_Client.cpp

```diff
@@ -0,0 +1,158 @@
+// Licensed under the MIT License.
+#include "WinGetServer.h"
+#include "Utils.h"
+
+#include <wil/com.h>
+#include <wil/result.h>
+#include <wil/safecast.h>
+
+#include <memory>
+#include <mutex>
+#include <string>
+#include <shtypes.h>
+#include <filesystem>
+#include <shlobj_core.h>
+
+#if USE_PROD_WINGET_SERVER
+const std::wstring_view s_ServerExePath = L"Microsoft\\WindowsApps\\Microsoft.DesktopAppInstaller_8wekyb3d8bbwe\\WindowsPackageManagerServer.exe";
+#else
+const std::wstring_view s_ServerExePath = L"Microsoft\\WindowsApps\\WinGetDevCLI_8wekyb3d8bbwe\\WindowsPackageManagerServerDev.exe";
+#endif
+
+_Must_inspect_result_
+_Ret_maybenull_ _Post_writable_byte_size_(size)
+void* __RPC_USER MIDL_user_allocate(_In_ size_t size)
+{
+    return malloc(size);
+}
+
+void __RPC_USER MIDL_user_free(_Pre_maybenull_ _Post_invalid_ void* ptr)
+{
+    if (ptr)
+    {
+        free(ptr);
+    }
+}
+
+std::filesystem::path GetKnownFolderPath(const KNOWNFOLDERID& id)
+{
+    wil::unique_cotaskmem_string knownFolder = nullptr;
+    THROW_IF_FAILED(SHGetKnownFolderPath(id, KF_FLAG_NO_ALIAS | KF_FLAG_DONT_VERIFY | KF_FLAG_NO_PACKAGE_REDIRECTION, NULL, &knownFolder));
+    return knownFolder.get();
+}
+
+struct FreeWithRpcStringFree { void operator()(RPC_CSTR* in) { RpcStringFreeA(in); } };
+using UniqueRpcString = std::unique_ptr<RPC_CSTR, FreeWithRpcStringFree>;
+
+struct DeleteWithMidlFree { void operator()(void* m) { MIDL_user_free(m); } };
+using UniqueMidl = std::unique_ptr<BYTE, DeleteWithMidlFree>;
+
+void InitializeRpcBinding()
+{
+    std::string protocol = "ncacn_np";
+    std::string endpoint = "\\pipe\\WinGetServerManualActivation_" + GetUserSID();
+
+    unsigned char* binding = nullptr;
+    UniqueRpcString bindingPtr;
+
+    RPC_STATUS status = RpcStringBindingComposeA(nullptr, GetUCharString(protocol), nullptr, GetUCharString(endpoint), nullptr, &binding);
+    THROW_HR_IF(HRESULT_FROM_WIN32(status), status != RPC_S_OK);
+    bindingPtr.reset(&binding);
+
+    status = RpcBindingFromStringBindingA(binding, &WinGetServerManualActivation_IfHandle);
+    THROW_HR_IF(HRESULT_FROM_WIN32(status), status != RPC_S_OK);
+}
+
+HRESULT LaunchWinGetServerWithManualActivation()
+{
+    const std::filesystem::path& localAppDataPath = GetKnownFolderPath(FOLDERID_LocalAppData);
+    const std::filesystem::path& serverExePath = localAppDataPath / s_ServerExePath;
+    std::wstring commandLineInput = std::wstring{ serverExePath } + L" --manualActivation";
+
+    STARTUPINFO info = { sizeof(info) };
+    wil::unique_process_information process;
+
+    RETURN_LAST_ERROR_IF(!CreateProcessW(NULL, &commandLineInput[0], NULL, NULL, FALSE, 0, NULL, NULL, &info, &process));
+
+    // Wait for manual reset event from server before proceeding with COM activation.
+    wil::unique_event manualResetEvent;
+    if (manualResetEvent.try_open(L"WinGetServerStartEvent"))
+    {
+        manualResetEvent.wait();
+        manualResetEvent.ResetEvent();

```

Comment:
>manualResetEvent.ResetEvent(); [](http://example.com/codeflow?start=8&length=30)

Do we need to reset event here? I thought the server could be connected by multiple clients?